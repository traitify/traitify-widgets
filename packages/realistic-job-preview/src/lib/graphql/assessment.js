export const create = `
  mutation createAssessment(
    $localeKey: String!,
    $profileID: String!,
    $surveyID: String!
  ) {
    createAssessment(
      localeKey: $localeKey,
      profileId: $profileID,
      surveyId: $surveyID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const get = `
  query assessment(
    $assessmentID: ID!
  ) {
    assessment(
      id: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const path = "/realistic-job-previews/graphql";

export const start = `
  mutation startAssessment(
    $assessmentID: ID!
  ) {
    startAssessment(
      id: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const update = `
  mutation updateAssessmentAnswer(
    $answers: [AssessmentAnswer],
    $assessmentID: String!
  ) {
    updateAssessmentAnswer(
      answers: $answers,
      assessmentId: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;
