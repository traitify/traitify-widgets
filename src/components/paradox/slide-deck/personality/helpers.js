export function completedSlides(slides) {
  return slides.filter(({likert_response: likertResponse, response}) => (
    likertResponse || response != null
  )).map(({id, likert_response: likertResponse, response, time_taken: timeTaken}) => ({
    id,
    likert_response: likertResponse,
    response,
    time_taken: timeTaken && timeTaken >= 0 ? timeTaken : 2
  }));
}

export function isFinished(slides) {
  return slides.length > 0 && slides.length === completedSlides(slides).length;
}

export function isFullscreen() {
  const fullscreen = document.fullscreenElement
    || document.webkitFullscreenElement
    || document.mozFullScreenElement
    || document.msFullscreenElement;

  return !!fullscreen;
}

export function loadingIndex(slides) {
  return slides.findIndex((slide) => (
    !slide.loaded && !slide.likert_response && slide.response == null
  ));
}

export function slideIndex(slides) {
  return slides.findIndex((slide) => (!slide.likert_response && slide.response == null));
}

export function toggleFullscreen(options) {
  const {current, element} = options;

  if(current) {
    const exitFullscreen = document.exitFullscreen
      || document.webkitExitFullscreen
      || document.mozCancelFullScreen
      || document.msExitFullscreen;

    exitFullscreen.apply(document);
  } else {
    const requestFullscreen = element.requestFullscreen
      || element.webkitRequestFullscreen
      || element.mozRequestFullScreen
      || element.msRequestFullscreen;

    requestFullscreen.apply(element);
  }
}

export function isReady(slides) {
  if(slides.length === 0) { return false; }

  const loadedIndex = loadingIndex(slides);
  const remainingSlidesLoaded = loadedIndex === -1;
  const nextSlidesLoaded = loadedIndex > slideIndex(slides) + 2;

  return remainingSlidesLoaded || nextSlidesLoaded;
}
