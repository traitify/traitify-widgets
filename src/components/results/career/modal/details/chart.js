import {useMemo} from "react";
import Chart from "react-apexcharts";
import useCareer from "lib/hooks/use-career";
import useResults from "lib/hooks/use-results";
import style from "./style.scss";

export default function CareerModalChart() {
  const career = useCareer();
  const results = useResults();
  const options = useMemo(() => {
    if(!career) { return; }
    if(!results) { return; }

    const traits = career.personality_traits
      .filter(({personality_trait: trait}) => trait)
      .map(({personality_trait: {id, name}, weight}, index) => {
        const trait = results.personality_traits.find((t) => id === t.personality_trait.id);
        if(!trait) { return; }

        return {
          assessment: Math.round((trait.score + 100) / 2),
          career: weight,
          name: name || `Trait ${index + 1}`
        };
      }).filter(Boolean).sort((a, b) => ((a.name > b.name) ? 1 : -1));

    return {
      height: 300,
      options: {
        fill: {opacity: 0.7},
        xaxis: {categories: traits.map(({name}) => name)},
        yaxis: {max: 100, show: false}
      },
      series: [
        {data: traits.map((trait) => trait.career), name: `${career.title} Traits`},
        {data: traits.map((trait) => trait.assessment), name: "Your Traits"}
      ],
      type: "radar"
    };
  }, [career, results]);

  if(!options) { return null; }

  return (
    <div className={style.chartContainer}>
      <Chart {...options} />
    </div>
  );
}
