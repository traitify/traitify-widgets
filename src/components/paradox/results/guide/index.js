import {faChevronDown, faChevronUp, faQuestion} from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import {useEffect, useState} from "react";
import {combine} from "lib/helpers/combine-data";
import {useDidMount, useDidUpdate} from "lib/helpers/hooks";
import Icon from "lib/helpers/icon";
import {dig} from "lib/helpers/object";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

function List({data}) {
  return (
    /* eslint-disable-next-line react/no-array-index-key */
    <ul>{data.split("\n").map((e, i) => <li key={i}>{e}</li>)}</ul>
  );
}

List.propTypes = {data: PropTypes.string.isRequired};

function Question({question, translate}) {
  const [showContent, setShowContent] = useState(question.order === 1);

  return (
    <div className={[style.question, showContent && style.open].filter(Boolean).join(" ")}>
      <div className={style.text}>
        <Icon className={style.icon} icon={faQuestion} />
        <div>{question.text}</div>
        <button onClick={() => setShowContent(!showContent)} type="button">
          <Icon className={style.icon} icon={showContent ? faChevronUp : faChevronDown} />
        </button>
      </div>
      {showContent && (
        <>
          <div className={style.h2}>{translate("question_purpose")}</div>
          <List data={question.purpose} />
          {question.adaptability && (
            <>
              <div className={style.h2}>{translate("question_adaptability")}</div>
              <List data={question.adaptability} />
            </>
          )}
        </>
      )}
    </div>
  );
}

Question.propTypes = {
  question: PropTypes.shape({
    adaptability: PropTypes.string,
    order: PropTypes.number.isRequired,
    purpose: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired
  }).isRequired,
  translate: PropTypes.func.isRequired
};

function Guide(props) {
  const {
    assessment,
    benchmark,
    element,
    followBenchmark,
    followGuide,
    guide,
    isReady,
    translate,
    ui
  } = props;
  const [activeCompetency, setActiveCompetency] = useState(null);
  const [data, setData] = useState([]);
  const [showExpandedIntro, setShowExpandedIntro] = useState(false);
  const state = {activeCompetency, showExpandedIntro};

  useDidMount(() => { ui.trigger("Guide.initialized", {props, state}); });
  useDidMount(() => { followBenchmark(); });
  useDidMount(() => { followGuide(); });
  useDidUpdate(() => { ui.trigger("Guide.updated", {props, state}); });
  useEffect(() => {
    const competencies = dig(guide, "competencies") || [];
    const types = dig(assessment, "personality_types") || [];
    if(competencies.length === 0 || types.length === 0) { return; }

    const _data = combine({benchmark, guide, types})
      .map(({competency, rank, score}) => ({...competency, rank, score}));

    setData(_data);
    setActiveCompetency(_data[0]);
  }, [
    dig(assessment, "id"),
    dig(benchmark, "id"),
    dig(guide, "assessment_id"),
    dig(guide, "locale_key")
  ]);

  if(!isReady("guide")) { return null; }
  if(!isReady("results")) { return null; }
  if(!activeCompetency) { return null; }

  const showCompetency = (newID) => setActiveCompetency(data.find(({id}) => newID === id));
  const [intro, ...expandedIntro] = activeCompetency.introduction.split("\n");
  const onChange = ({target: {value}}) => showCompetency(value);
  const index = data.indexOf(activeCompetency);
  const contentClass = [
    style.content,
    index === 0 && style.first,
    index === data.length - 1 && style.last
  ].filter(Boolean).join(" ");

  return (
    <div className={style.container} ref={element}>
      <div className={style.tabs}>
        {data.map(({id, name, rank}) => (
          <button
            key={id}
            className={[id === activeCompetency.id && style.active, style[rank]].filter(Boolean).join(" ")}
            onClick={() => showCompetency(id)}
            type="button"
          >
            <span>{name}</span>
          </button>
        ))}
      </div>
      <div className={contentClass}>
        <select className={style.dropdown} onChange={onChange} value={activeCompetency.id}>
          {data.map(({id, name}) => <option key={id} value={id}>{name}</option>)}
        </select>
        <div className={[style.heading, style[activeCompetency.rank]].join(" ")}>{activeCompetency.name}</div>
        {intro}
        <p>
          <button className={style.readMore} onClick={() => setShowExpandedIntro(!showExpandedIntro)} type="button">
            {translate("read_more")}
          </button>
        </p>
        {showExpandedIntro && <p>{expandedIntro.join("\n").trim()}</p>}
        <hr />
        {activeCompetency.questionSequences.map((sequence) => (
          <div key={sequence.id}>
            <div className={style.heading}>{sequence.name}</div>
            <p>{translate("guide_intro")}</p>
            {sequence.questions.map((question) => (
              <Question key={question.id} question={question} translate={translate} />
            ))}
          </div>
        ))}
      </div>
    </div>
  );
}

Guide.defaultProps = {assessment: null, benchmark: null, element: null, guide: null};
Guide.propTypes = {
  assessment: PropTypes.shape({
    assessment_type: PropTypes.string,
    personality_types: PropTypes.arrayOf(
      PropTypes.shape({
        personality_type: PropTypes.shape({id: PropTypes.string.isRequired}).isRequired,
        score: PropTypes.number.isRequired
      }).isRequired
    )
  }),
  benchmark: PropTypes.shape({
    id: PropTypes.string,
    range_types: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        ranges: PropTypes.arrayOf(
          PropTypes.shape({
            match_score: PropTypes.number.isRequired,
            max_score: PropTypes.number.isRequired,
            min_score: PropTypes.number.isRequired
          }).isRequired
        ).isRequired
      }).isRequired
    )
  }),
  element: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({current: PropTypes.instanceOf(Element)})
  ]),
  followBenchmark: PropTypes.func.isRequired,
  followGuide: PropTypes.func.isRequired,
  guide: PropTypes.shape({
    assessment_id: PropTypes.string.isRequired,
    competencies: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        questionSequences: PropTypes.arrayOf(
          PropTypes.shape({
            id: PropTypes.string.isRequired,
            name: PropTypes.string.isRequired,
            personalityTypeId: PropTypes.string.isRequired,
            questions: PropTypes.arrayOf(
              PropTypes.shape({
                id: PropTypes.string.isRequired
              }).isRequired
            ).isRequired
          }).isRequired
        ).isRequired
      }).isRequired
    ).isRequired,
    locale_key: PropTypes.string.isRequired
  }),
  isReady: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {Guide as Component};
export default withTraitify(Guide);
