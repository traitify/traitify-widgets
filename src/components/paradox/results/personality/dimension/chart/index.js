import PropTypes from "prop-types";
import {reverse} from "lib/helpers/array";
import {combine} from "lib/helpers/combine-data";
import {useDidMount, useDidUpdate} from "lib/helpers/hooks";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

const ranks = ["other", "low", "medium", "high"];

function PersonalityDimensionChart(props) {
  const {assessment, element, followGuide, guide, isReady, translate, ui} = props;
  const state = {};

  useDidMount(() => { ui.trigger("PersonalityDimensionChart.initialized", {props, state}); });
  useDidMount(() => { ui.trigger("PersonalityDimensionColumns.initialized", {props, state}); });
  useDidMount(() => { followGuide(); });
  useDidUpdate(() => { ui.trigger("PersonalityDimensionChart.updated", {props, state}); });
  useDidUpdate(() => { ui.trigger("PersonalityDimensionColumns.updated", {props, state}); });

  if(!isReady("results")) { return null; }

  const data = combine({guide, order: "types", types: assessment.personality_types});

  return (
    <div className={style.container} ref={element}>
      <div className={style.p}>{translate("dimension_description")}</div>
      <div className={style.horizontal}>
        {data.map(({competency, rank, type: {badge, name, id}}) => (
          <div key={id} className={[style.row, style[rank]].join(" ")}>
            <div className={style.label}>
              {competency && <div>{competency.name}</div>}
              <div>{name}</div>
            </div>
            <img src={badge.image_medium} alt={`${name} ${translate("badge")}`} />
          </div>
        ))}
        <div className={style.scale}>
          {ranks.map((rank) => <div key={rank} />)}
        </div>
      </div>
      <div className={style.vertical}>
        <div className={style.scale}>
          {reverse(ranks).map((rank) => <div key={rank} />)}
        </div>
        {data.map(({competency, rank, type: {badge, name, id}}) => (
          <div key={id} className={[style.column, style[rank]].join(" ")}>
            <img src={badge.image_medium} alt={`${name} ${translate("badge")}`} />
            {competency && <div className={style.heading}>{competency.name}</div>}
            <div className={style.heading}>{name}</div>
          </div>
        ))}
      </div>
    </div>
  );
}

PersonalityDimensionChart.defaultProps = {assessment: null, element: null, guide: null};
PersonalityDimensionChart.propTypes = {
  assessment: PropTypes.shape({
    personality_types: PropTypes.arrayOf(
      PropTypes.shape({
        personality_type: PropTypes.shape({
          badge: PropTypes.shape({image_medium: PropTypes.string.isRequired}).isRequired,
          id: PropTypes.string.isRequired,
          name: PropTypes.string.isRequired
        }).isRequired,
        score: PropTypes.number.isRequired
      }).isRequired
    )
  }),
  element: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({current: PropTypes.instanceOf(Element)})
  ]),
  followGuide: PropTypes.func.isRequired,
  guide: PropTypes.shape({
    assessment_id: PropTypes.string.isRequired,
    competencies: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        questionSequences: PropTypes.arrayOf(
          PropTypes.shape({
            id: PropTypes.string.isRequired,
            name: PropTypes.string.isRequired,
            personalityTypeId: PropTypes.string.isRequired,
            questions: PropTypes.arrayOf(
              PropTypes.shape({
                id: PropTypes.string.isRequired
              }).isRequired
            ).isRequired
          }).isRequired
        ).isRequired
      }).isRequired
    ).isRequired,
    locale_key: PropTypes.string.isRequired
  }),
  isReady: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {PersonalityDimensionChart as Component};
export default withTraitify(PersonalityDimensionChart);
