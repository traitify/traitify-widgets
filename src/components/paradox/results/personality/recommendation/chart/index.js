import PropTypes from "prop-types";
import {useEffect, useState} from "react";
import {reverse, times} from "lib/helpers/array";
import {createColumns} from "lib/helpers/combine-data";
import {dig} from "lib/helpers/object";
import TraitifyPropTypes from "lib/helpers/prop-types";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

// const ranks = [
//   {key: "preferred", rank: "high"},
//   {key: "acceptable", rank: "medium"},
//   {key: "potential_risk", rank: "low"}
// ];
const colors = {high: "#29B770", low: "#EF615E", medium: "#FFCC3B", other: "black"};

function PersonalityRecommendationChart({setElement, ...props}) {
  const {
    assessment,
    benchmark,
    combined,
    followBenchmark,
    followGuide,
    getOption,
    guide,
    isReady,
    // translate,
    ui
  } = props;
  const [data, setData] = useState([]);
  const state = {};

  useDidMount(() => { ui.trigger("PersonalityRecommendationChart.initialized", {props, state}); });
  useDidMount(() => { followBenchmark(); });
  useDidMount(() => { followGuide(); });
  useDidUpdate(() => { ui.trigger("PersonalityRecommendationChart.updated", {props, state}); });
  useEffect(() => {
    const competencies = dig(guide, "competencies") || [];
    const types = dig(assessment, "personality_types") || [];
    if(competencies.length === 0 || types.length === 0) { return; }

    setData(createColumns({benchmark, guide, order: "types", types}));
  }, [
    dig(assessment, "personality_types", 0, "personality_type", "name"),
    dig(benchmark, "id"),
    dig(benchmark, "rankings", 0, "description"),
    dig(guide, "assessment_id"),
    dig(guide, "locale_key")
  ]);

  const disabledComponents = getOption("disabledComponents") || [];
  if(disabledComponents.includes("PersonalityRecommendationChart")) { return null; }
  if(!isReady("guide")) { return null; }
  if(!isReady("results")) { return null; }
  if(data.length === 0) { return null; }

  const length = Math.max(...data.map(({data: points}) => points.length));

  const colorFromRank = (rank) => {
    if(rank === "low") { return benchmark ? benchmark.hexColorLow : colors.low; }
    if(rank === "medium") { return benchmark ? benchmark.hexColorMedium : colors.medium; }
    if(rank === "high") { return benchmark ? benchmark.hexColorHigh : colors.high; }

    return colors.other;
  };

  return (
    <div className={[style.container, combined && style.combined].filter(Boolean).join(" ")} ref={setElement}>
      {/* <div className={style.ranks}>
        {ranks.map(({key, rank}) => <div key={key} className={style[rank]}>{translate(key)}</div>)}
      </div> */}
      <div className={style.vertical}>
        <div className={style.scale}>
          {times(length).map((index) => (
            <div key={index} className={style.label}>
              <span>{length - index}</span>
            </div>
          ))}
        </div>
        {data.map(({competency, data: points, type}) => (
          <div key={type.id} className={style.column}>
            {reverse(points).map(({active, rank}, index) => (
              <div
                key={index} /* eslint-disable-line react/no-array-index-key */
                className={[active && style.active, style[rank]].filter(Boolean).join(" ")}
                style={{background: colorFromRank(rank)}}
              />
            ))}
            <div className={style.name}>{competency.name}</div>
          </div>
        ))}
      </div>
      <div className={style.horizontal}>
        <div className={style.labels}>
          {data.map(({competency, type}) => <div key={type.id}>{competency.name}</div>)}
        </div>
        <div className={style.chart}>
          <div className={style.grid}>
            {data.map(({data: points, type}) => (
              <div key={type.id} className={style.row}>
                {points.map(({active, rank}, index) => (
                  <div
                    key={index} /* eslint-disable-line react/no-array-index-key */
                    className={[active && style.active, style[rank]].filter(Boolean).join(" ")}
                    style={{background: colorFromRank(rank)}}
                  />
                ))}
              </div>
            ))}
          </div>
          <div className={style.scale}>
            {times(length).map((index) => <div key={index}>{index + 1}</div>)}
          </div>
        </div>
      </div>
    </div>
  );
}

PersonalityRecommendationChart.defaultProps = {
  assessment: null,
  benchmark: null,
  combined: false,
  guide: null
};
PersonalityRecommendationChart.propTypes = {
  assessment: PropTypes.shape({
    personality_types: PropTypes.arrayOf(
      PropTypes.shape({
        personality_type: PropTypes.shape({
          name: PropTypes.string.isRequired
        }).isRequired
      }).isRequired
    )
  }),
  benchmark: PropTypes.shape({
    id: PropTypes.string,
    hexColorLow: PropTypes.string.isRequired,
    hexColorMedium: PropTypes.string.isRequired,
    hexColorHigh: PropTypes.string.isRequired,
    dimensionRanges: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        dimensionId: PropTypes.string.isRequired,
        matchScore: PropTypes.number.isRequired,
        maxScore: PropTypes.number.isRequired,
        minScore: PropTypes.number.isRequired
      }).isRequired
    ),
    resultRankings: PropTypes.arrayOf(
      PropTypes.shape({
        description: PropTypes.string.isRequired
      }).isRequired
    )
  }),
  combined: PropTypes.bool,
  followBenchmark: PropTypes.func.isRequired,
  followGuide: PropTypes.func.isRequired,
  getOption: PropTypes.func.isRequired,
  guide: PropTypes.shape({
    assessment_id: PropTypes.string.isRequired,
    competencies: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      }).isRequired
    ).isRequired,
    locale_key: PropTypes.string.isRequired
  }),
  isReady: PropTypes.func.isRequired,
  setElement: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};

export {PersonalityRecommendationChart as Component};
export default withTraitify(PersonalityRecommendationChart);
