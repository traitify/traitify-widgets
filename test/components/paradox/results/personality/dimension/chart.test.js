import {Component} from "components/paradox/results/personality/dimension/chart";
import ComponentHandler from "support/component-handler";
import {mockProps} from "support/helpers";
import assessment from "support/json/assessment/dimension-based.json";
import guide from "support/json/guide.json";

jest.mock("lib/with-traitify", () => ((value) => value));

describe("Paradox.PersonalityDimensionChart", () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      ...mockProps(["followGuide", "isReady", "setElement", "translate", "ui"]),
      assessment,
      guide: {
        assessment_id: assessment.id,
        locale_key: assessment.locale_key,
        ...guide
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger).toHaveBeenCalledWith(
        "PersonalityDimensionChart.initialized",
        expect.objectContaining({
          props: expect.any(Object),
          state: expect.any(Object)
        })
      );
      expect(props.ui.trigger).toHaveBeenCalledWith(
        "PersonalityDimensionColumns.initialized",
        expect.objectContaining({
          props: expect.any(Object),
          state: expect.any(Object)
        })
      );
    });

    it("triggers update", () => {
      component = new ComponentHandler(<Component {...props} />);
      props.ui.trigger.mockClear();
      component.updateProps();

      expect(props.ui.trigger).toHaveBeenCalledWith(
        "PersonalityDimensionChart.updated",
        expect.objectContaining({
          props: expect.any(Object),
          state: expect.any(Object)
        })
      );
      expect(props.ui.trigger).toHaveBeenCalledWith(
        "PersonalityDimensionColumns.updated",
        expect.objectContaining({
          props: expect.any(Object),
          state: expect.any(Object)
        })
      );
    });
  });

  it("renders component", () => {
    component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component without guide", () => {
    props.guide = null;
    props.isReady.mockImplementation((value) => value !== "guide");
    component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if results not ready", () => {
    props.assessment = null;
    props.isReady.mockImplementation((value) => value !== "results");
    component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });
});
