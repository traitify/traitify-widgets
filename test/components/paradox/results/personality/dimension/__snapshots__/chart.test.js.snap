// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paradox.PersonalityDimensionChart renders component 1`] = `
<div
  className="container"
>
  <div
    className="p"
  >
    dimension_description
  </div>
  <div
    className="horizontal"
  >
    <div
      className="row high"
    >
      <div
        className="label"
      >
        <div>
          Influencing People
        </div>
        <div>
          Agreeableness
        </div>
      </div>
      <img
        alt="Agreeableness badge"
        src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
      />
    </div>
    <div
      className="row medium"
    >
      <div
        className="label"
      >
        <div>
          Engaging with People
        </div>
        <div>
          Extraversion
        </div>
      </div>
      <img
        alt="Extraversion badge"
        src="https://cdn.traitify.com/badges/bf796c03-a863-4f61-9a1a-1f4fe6e408e8_medium"
      />
    </div>
    <div
      className="row medium"
    >
      <div
        className="label"
      >
        <div>
          Delivering Results
        </div>
        <div>
          Conscientiousness
        </div>
      </div>
      <img
        alt="Conscientiousness badge"
        src="https://cdn.traitify.com/badges/0d29bdb5-b502-4036-a790-9479d588e8fb_medium"
      />
    </div>
    <div
      className="row medium"
    >
      <div
        className="label"
      >
        <div>
          Solving Problems
        </div>
        <div>
          Openness
        </div>
      </div>
      <img
        alt="Openness badge"
        src="https://cdn.traitify.com/badges/216e23f2-f27c-46ae-b4b1-00c97a01db22_medium"
      />
    </div>
    <div
      className="row low"
    >
      <div
        className="label"
      >
        <div>
          Managing Pressure
        </div>
        <div>
          Emotional Stability
        </div>
      </div>
      <img
        alt="Emotional Stability badge"
        src="https://cdn.traitify.com/badges/16cd30cc-3d5c-4c7e-b13f-828853034334_medium"
      />
    </div>
    <div
      className="scale"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
  <div
    className="vertical"
  >
    <div
      className="scale"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="column high"
    >
      <img
        alt="Agreeableness badge"
        src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
      />
      <div
        className="heading"
      >
        Influencing People
      </div>
      <div
        className="heading"
      >
        Agreeableness
      </div>
    </div>
    <div
      className="column medium"
    >
      <img
        alt="Extraversion badge"
        src="https://cdn.traitify.com/badges/bf796c03-a863-4f61-9a1a-1f4fe6e408e8_medium"
      />
      <div
        className="heading"
      >
        Engaging with People
      </div>
      <div
        className="heading"
      >
        Extraversion
      </div>
    </div>
    <div
      className="column medium"
    >
      <img
        alt="Conscientiousness badge"
        src="https://cdn.traitify.com/badges/0d29bdb5-b502-4036-a790-9479d588e8fb_medium"
      />
      <div
        className="heading"
      >
        Delivering Results
      </div>
      <div
        className="heading"
      >
        Conscientiousness
      </div>
    </div>
    <div
      className="column medium"
    >
      <img
        alt="Openness badge"
        src="https://cdn.traitify.com/badges/216e23f2-f27c-46ae-b4b1-00c97a01db22_medium"
      />
      <div
        className="heading"
      >
        Solving Problems
      </div>
      <div
        className="heading"
      >
        Openness
      </div>
    </div>
    <div
      className="column low"
    >
      <img
        alt="Emotional Stability badge"
        src="https://cdn.traitify.com/badges/16cd30cc-3d5c-4c7e-b13f-828853034334_medium"
      />
      <div
        className="heading"
      >
        Managing Pressure
      </div>
      <div
        className="heading"
      >
        Emotional Stability
      </div>
    </div>
  </div>
</div>
`;

exports[`Paradox.PersonalityDimensionChart renders component without guide 1`] = `
<div
  className="container"
>
  <div
    className="p"
  >
    dimension_description
  </div>
  <div
    className="horizontal"
  >
    <div
      className="row high"
    >
      <div
        className="label"
      >
        <div>
          Agreeableness
        </div>
      </div>
      <img
        alt="Agreeableness badge"
        src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
      />
    </div>
    <div
      className="row medium"
    >
      <div
        className="label"
      >
        <div>
          Extraversion
        </div>
      </div>
      <img
        alt="Extraversion badge"
        src="https://cdn.traitify.com/badges/bf796c03-a863-4f61-9a1a-1f4fe6e408e8_medium"
      />
    </div>
    <div
      className="row medium"
    >
      <div
        className="label"
      >
        <div>
          Conscientiousness
        </div>
      </div>
      <img
        alt="Conscientiousness badge"
        src="https://cdn.traitify.com/badges/0d29bdb5-b502-4036-a790-9479d588e8fb_medium"
      />
    </div>
    <div
      className="row medium"
    >
      <div
        className="label"
      >
        <div>
          Openness
        </div>
      </div>
      <img
        alt="Openness badge"
        src="https://cdn.traitify.com/badges/216e23f2-f27c-46ae-b4b1-00c97a01db22_medium"
      />
    </div>
    <div
      className="row low"
    >
      <div
        className="label"
      >
        <div>
          Emotional Stability
        </div>
      </div>
      <img
        alt="Emotional Stability badge"
        src="https://cdn.traitify.com/badges/16cd30cc-3d5c-4c7e-b13f-828853034334_medium"
      />
    </div>
    <div
      className="scale"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
  <div
    className="vertical"
  >
    <div
      className="scale"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="column high"
    >
      <img
        alt="Agreeableness badge"
        src="https://cdn.traitify.com/badges/96dbf653-6ea0-4dc0-94dc-f6db3247510c_medium"
      />
      <div
        className="heading"
      >
        Agreeableness
      </div>
    </div>
    <div
      className="column medium"
    >
      <img
        alt="Extraversion badge"
        src="https://cdn.traitify.com/badges/bf796c03-a863-4f61-9a1a-1f4fe6e408e8_medium"
      />
      <div
        className="heading"
      >
        Extraversion
      </div>
    </div>
    <div
      className="column medium"
    >
      <img
        alt="Conscientiousness badge"
        src="https://cdn.traitify.com/badges/0d29bdb5-b502-4036-a790-9479d588e8fb_medium"
      />
      <div
        className="heading"
      >
        Conscientiousness
      </div>
    </div>
    <div
      className="column medium"
    >
      <img
        alt="Openness badge"
        src="https://cdn.traitify.com/badges/216e23f2-f27c-46ae-b4b1-00c97a01db22_medium"
      />
      <div
        className="heading"
      >
        Openness
      </div>
    </div>
    <div
      className="column low"
    >
      <img
        alt="Emotional Stability badge"
        src="https://cdn.traitify.com/badges/16cd30cc-3d5c-4c7e-b13f-828853034334_medium"
      />
      <div
        className="heading"
      >
        Emotional Stability
      </div>
    </div>
  </div>
</div>
`;

exports[`Paradox.PersonalityDimensionChart renders nothing if results not ready 1`] = `null`;
